jwst/datamodels/utils/flat_multispec.py

```diff
@@ -165,6 +171,26 @@ def populate_recarray(output_table, input_spec, n_rows, columns, is_vector, igno
         log.warning(f"Metadata could not be determined from input spec_table: {problems}")
 
 
+def set_schema_units(model):
```

Comment:
@melanieclarke how do you feel about the way I handled default units?  If we like it, this could be extended to other parts of extract_1d or maybe elsewhere in the repository.  Granted, it's a band-aid for stdatamodels issues, but I don't know how long it'll be before we can make the required stdatamodels changes.